File: 프로그래머스/1/161989. 덧칠하기/덧칠하기.java
import java.util.*;

class Solution {
    public int solution(int n, int m, int[] section) {
        int answer = 0;
        
        List<Integer> arr = new ArrayList<>();
        
        for(int i : section){
            arr.add(i);
        }
        
        for(int i = 0; i < section.length; i++){
            int num = section[i];
            if(arr.contains(num)){
                answer++;
                for(int j = num + m - 1; j >= num; j--){
                    if(arr.contains(j)){
                        arr.remove(Integer.valueOf(j));
                    }
                }
            }
        }
        
        return answer;
    }
}

File: 프로그래머스/1/161989. 덧칠하기/덧칠하기.swift
import Foundation

func solution(_ n:Int, _ m:Int, _ section:[Int]) -> Int {
    var answer = 0
    var start = section[0]
    answer += 1
    for i in section {
        if start + m > i {
            continue
        }
        start = i
        answer += 1
    }
    
    return answer
}